power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/common.hpp

```diff
@@ -46,6 +46,11 @@ reordered_attribute_buffers(BufferType& buffer, std::span<MetaAttribute const* c
             }
             return AttributeBuffer<void>{};
         });
+
+    if (std::ranges::all_of(result, [](auto const& attribute_buffer) { return attribute_buffer.data == nullptr; })) {
+        result = {};
```

Comment:
this is the pre-calculate part that checks whether the intersection is empty between
* the set of all attributes listed in the serialized data header for array-like elements; and
* the set of all attribute buffers provided by the user

Reply:
Please put an in-code documentation here to specify the intention.

Reply:
added docstring